File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    man1, man2, man3 = 0, 0, 0
    man1_rule = [1, 2, 3, 4, 5]
    man2_rule = [2, 1, 2, 3, 2, 4, 2, 5]
    man3_rule = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]

    for i in range(len(answers)):
        if answers[i] == man1_rule[i%len(man1_rule)]:
            man1 += 1
        if answers[i] == man2_rule[i%len(man2_rule)]:
            man2 += 1
        if answers[i] == man3_rule[i%len(man3_rule)]:
            man3 += 1
    arr = [man1, man2, man3]
    max_num = max(arr)
    for i in range(len(arr)):
        if max_num <= arr[i]:
            max_num = arr[i]
            answer.append(i + 1)
    return answer

File: 프로그래머스/1/42840. 모의고사/모의고사.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> answers) {
    vector<int> answer;
    
    vector<int> man1 = {1, 2, 3, 4, 5};
    vector<int> man2 = {2, 1, 2, 3, 2, 4, 2, 5};
    vector<int> man3 = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
    
    int man1_pt = 0;
    int man2_pt = 0;
    int man3_pt = 0;
    for (int i = 0; i < answers.size(); i++) {
        if (answers[i] == man1[i % man1.size()]) {
            man1_pt++;
        }
        if (answers[i] == man2[i % man2.size()]) {
            man2_pt++;
        }
        if (answers[i] == man3[i % man3.size()]) {
            man3_pt++;
        }
    }
    vector<int> pts = {man1_pt, man2_pt, man3_pt};
    int best = *max_element(pts.begin(), pts.end());
    for (int i = 0; i < pts.size(); i++) {
        if (best <= pts[i]) {
            answer.push_back(i + 1);
        }
    }
    return answer;
}